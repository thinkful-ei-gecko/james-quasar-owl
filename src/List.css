.participant-list {

}
.participant-list li {
  display: flex; 
  border: 1px solid red;
}

.participant-avatar {
  width: 50px;
  height: 50px;
  border-radius: 100%;
}

.participant-details p {
}

.participant-details h2 {
  font-size: 18px;
  font-weight: 600;
}

.active p{
  display: flex;
  align-items: center;
  font-size: 1em;
}

.inactive p{
  display: flex;
  align-items: center;
  font-size: 1em;
}

.active p::before {
  color: rgb(59, 216, 59);
  content: '•';
  font-size: 1.5rem;
  margin-right: .3rem;
  display: inline-block;

}

.inactive p::before{
  color: grey;
  content: '•';
  font-size: 1.5rem;
  margin-right: .3rem;
}