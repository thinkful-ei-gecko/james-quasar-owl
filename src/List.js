import React from 'react';
import store from './store.js';
// import ReactDOM from 'react-dom';

function List (props) {
  const activeParticipantsList = store.participants.filter(person => {
    return person.inSession;
  });

  console.log(activeParticipantsList);

  const inActiveParticipantsList = store.participants.filter(person => {
    return !person.inSession;
  });

  const allParticipantsList = activeParticipantsList.concat(inActiveParticipantsList);

  return (
    <ul>
      {
        allParticipantsList.map( person => (
          <li key={person.id}>
            <img src={person.avatar} alt={person.name + ' avatar'}/>
            <div>
              <h2>{person.name}</h2>
              {person.inSession ? person.onStage ? <div className="active"><p>on stage</p></div> : <div className="active"><p>in session</p></div>: <div className="inactive"><p>left session</p></div>}
            </div>
          </li>
        ))}
    </ul>
  );
};

export default List;