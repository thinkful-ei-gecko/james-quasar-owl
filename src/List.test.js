import React from 'react';
import ReactDOM from 'react-dom';
import List from './List.js';
import renderer from 'react-test-renderer';
import { exportAllDeclaration } from '@babel/types';


describe('List test', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<List key={1}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders same UI', () => {
    const tree = renderer.create(<List key={1} />).toJSON();
    expect(tree).toMatchSnapshot();
  });



});