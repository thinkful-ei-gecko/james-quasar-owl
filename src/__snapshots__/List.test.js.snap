// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List test renders same UI 1`] = `
<ul>
  <li>
    <img
      alt="Koren Templeton avatar"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
    <div>
      <h2>
        Koren Templeton
      </h2>
      <div
        className="active"
      >
        <p>
          on stage
        </p>
      </div>
    </div>
  </li>
  <li>
    <img
      alt="Caty Flucker avatar"
      src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
    />
    <div>
      <h2>
        Caty Flucker
      </h2>
      <div
        className="active"
      >
        <p>
          in session
        </p>
      </div>
    </div>
  </li>
  <li>
    <img
      alt="Frank Runciman avatar"
      src="https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1"
    />
    <div>
      <h2>
        Frank Runciman
      </h2>
      <div
        className="active"
      >
        <p>
          in session
        </p>
      </div>
    </div>
  </li>
  <li>
    <img
      alt="Ashla Attwool avatar"
      src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
    />
    <div>
      <h2>
        Ashla Attwool
      </h2>
      <div
        className="active"
      >
        <p>
          on stage
        </p>
      </div>
    </div>
  </li>
  <li>
    <img
      alt="Axe Kubicka avatar"
      src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
    />
    <div>
      <h2>
        Axe Kubicka
      </h2>
      <div
        className="inactive"
      >
        <p>
          left session
        </p>
      </div>
    </div>
  </li>
</ul>
`;
